contracts/implementation/PoolFactory.sol

```diff
@@ -136,6 +143,10 @@ contract PoolFactory is IPoolFactory, ITwoStepGovernance {
             "Decimal precision too high"
         );
 
+        if (deploymentFee != 0) {
+            IERC20(deploymentFeeToken).transferFrom(msg.sender, deploymentFeeReceiver, deploymentFee);
+        }
+
```

Comment:
Normally this would be suspicious because of a lack of CEI adherence. However, since we will presumably be setting `deploymentFee` and `deploymentFeeReceiver` safely, I think it's fine.